player/http_handlers.go

```diff
@@ -51,6 +55,12 @@ func (h RequestHandler) processStreamError(w http.ResponseWriter, uri string, er
 		h.writeErrorResponse(w, http.StatusServiceUnavailable, err.Error())
 	} else if strings.Contains(err.Error(), "hash in response does not match") {
 		h.writeErrorResponse(w, http.StatusServiceUnavailable, err.Error())
+	} else if strings.Contains(err.Error(), "token contains an invalid number of segments") {
```

Comment:
where do these errors come from? this function makes me sad. whatever's sending these errors should be updated to use error codes so we're not relying on string matching

Reply:
That's from [jwt-go](https://github.com/dgrijalva/jwt-go) library. Perhaps I should create a pull request to improve this. They should be using error objects

Reply:
Actually it seems like they do have error codes. I just need to look better